examples/resources/spectrocloud_virtual_machine/resource.tf

```diff
@@ -1,51 +1,72 @@
+data "spectrocloud_cluster" "vm-enabled-cluster" {
+  name = "newyork"
+  context = "tenant"
+}
+
 resource "spectrocloud_virtual_machine" "tf-test-vm-default" {
-  cluster_uid = "6406b8ee7c865766fde23247"
+  cluster_uid = data.spectrocloud_cluster.vm-enabled-cluster.id //"6406b8ee7c865766fde23247"
   name = "tf-test-vm-default"
   namespace = "default"
-  cpu_cores = 2
-  memory = "3G"
+  cpu_cores = 1
+  memory = "2G"
   image_url = "quay.io/kubevirt/fedora-cloud-container-disk-demo"
-  vm_action = "migrate"
-  labels = ["tf-test=true", "env=dev"]
+  labels = ["tf-test=true", "env=sumit-dev"]
   annotations = {
     tf : "true",
-    owner: "siva",
+    owner: "siva1",
   }
 }
 
+resource "spectrocloud_virtual_machine" "tf-test-vm-clone-default" {
+  cluster_uid = data.spectrocloud_cluster.vm-enabled-cluster.id
+  clone_name = spectrocloud_virtual_machine.tf-test-vm-default.name
```

Comment:
seems like confusing name of property. I'd suggest `base_vm_name` so we won't confuse it with result of clone operation or a template attribute. Let's also ask @karl-cardenas-coding to check this naming.

Reply:
@nikchern I would defer to @cloudmaniac , he has a strong background in VMware and is helping create the initial docs for this feature.

Reply:
`clone_name` represents the source VM to clone? Then yes, it's confusing. `base_vm_name` could be fine, except if you use the same resource type to deploy from template AND to clone. If the latter is the case, we must be more generic.

In the vSphere provider, they use the same resource, and add a `clone` block ([doc](https://registry.terraform.io/providers/hashicorp/vsphere/latest/docs/resources/virtual_machine#cloning-and-customization)) with a reference another VM (or template ID).

`template_uuid = data.vsphere_virtual_machine.template.id`

_Note: I compare to the vSphere TF provider as the target is still the vSphere admins that may be used to that provider._

Reply:
rename to `base_vm_name`